pkg/synccompactor/naive/naive.go

```diff
@@ -0,0 +1,83 @@
+package naive
+
+import (
+	"context"
+
+	"github.com/conductorone/baton-sdk/pkg/dotc1z"
+	"google.golang.org/protobuf/proto"
+)
+
+func NewNaiveCompactor(base *dotc1z.C1File, applied *dotc1z.C1File, dest *dotc1z.C1File) *Compactor {
+	return &Compactor{
+		base:    base,
+		applied: applied,
+		dest:    dest,
+	}
+}
+
+type Compactor struct {
+	base    *dotc1z.C1File
+	applied *dotc1z.C1File
+	dest    *dotc1z.C1File
+}
+
+func (n *Compactor) Compact(ctx context.Context) error {
+	if err := n.processResourceTypes(ctx); err != nil {
+		return err
+	}
+	if err := n.processResources(ctx); err != nil {
+		return err
+	}
+	if err := n.processEntitlements(ctx); err != nil {
+		return err
+	}
+	if err := n.processGrants(ctx); err != nil {
```

Comment:
Partial syncs don't expand grants or external IDs, so that will need to be done once everything's compacted.

Reply:
That's ok, this should support any kind of sync on the applied side. This just copies the data from the tables over. But yes running grant expansion seems valuable.

Maybe we can just do that on the final compacted result?